tests/unit/lib/bootstrap/companion_stack/test_data_types.py

```diff
@@ -0,0 +1,63 @@
+from samcli.lib.bootstrap.companion_stack.data_types import CompanionStack, ECRRepo
+from samcli.lib.bootstrap.companion_stack.companion_stack_builder import CompanionStackBuilder
+from unittest import TestCase
+from unittest.mock import Mock, patch
+
+
+class TestCompanionStack(TestCase):
+    def setUp(self):
+        self.check_sum = "checksum"
+        self.parent_stack_name = "Parent-Stack"
+        self.check_sum_patch = patch("samcli.lib.bootstrap.companion_stack.data_types.str_checksum")
```

Comment:
Not sure this should be mocked.
If we were to change the checksum algorithm (`str_checksum` return a different value), I think we would like to see these tests fail to surface the change of name of our stacks.

Also, could you add some tests to verify the max-length cutting of logical id and checksum?

Reply:
Added tests for str_checksum and cutoff.